src/entity-manager.js

```diff
@@ -21,35 +21,36 @@ export class EntityManager {
   }
 
   /**
-   * Register an array of entity references.
+   * Register an array of entity classes.
    *
-   * @param {Entity[]|Entity} entities Array or object of entities.
+   * @param {function[]|function} Entity classes array or object of Entity constructors.
    *
    * @return {EntityManager} this
    * @chainable
    */
-  registerEntities(entities) {
-    for (let reference in entities) {
-      if (!entities.hasOwnProperty(reference)) {
+  registerEntities(EntityClasses) {
+    for (let property in EntityClasses) {
+      if (!EntityClasses.hasOwnProperty(property)
```

Comment:
Also, why not use instanceof? That's less confusing and exists for that purpose.